src/routes/api/email/+server.ts

```diff
@@ -0,0 +1,38 @@
+import mailchimp from '@mailchimp/mailchimp_transactional';
+import { env } from '$env/dynamic/private';
+import dynamics from '$lib/dynamics';
+import type { MessageRecipient } from '@mailchimp/mailchimp_transactional';
+
+const apiKey = env.MAILCHIMP_API_KEY;
+const client = mailchimp(apiKey ?? '');
+
+export async function POST({ request }) {
+	try {
+		const { email, results } = await request.json();
+		// todo: include email validation
+		const features = Object.keys(results);
+
+		const message = {
+			to: [{ email, type: 'to' } as MessageRecipient],
+			from_email: 'info@allwecansave.earth',
+			subject: '8 Dynamics - Your Results',
+			html: `<h1>Your Results: </h1>
+				<div>
+				${dynamics.map((d, i) => `<span>${d.short}: ${results[features[i]]}</span>`)}
+				</div>
+			`
+		};
+
+		const response = await client.messages.send({ message });
+		if (Array.isArray(response)) {
+			const result = response[0];
+			if (result.status === 'sent') {
+				return new Response(JSON.stringify({ success: true }), { status: 200 });
+			}
+		} else {
+			throw new Error(response.message);
+		}
```

Comment:
nit: I think we can run into returning nothing if  `result.status !== 'sent'`, could remove the `throw new Error(response.message);` outside of the `else` if we want an error to be returned in that case

Reply:
hmm yeah I see your point. this is a bit tricky bc of typing. 
<img width="1163" alt="Screenshot 2024-12-18 at 6 45 47 PM" src="https://github.com/user-attachments/assets/649632b7-885f-4bcd-92bd-7dd97e24414f" />

I've added an else case within the check of whether the result is is an array, and moved the original error throwing outside of the else case

Reply:
so now we throw an error if there wasn't an error but the email didn't send successfully, or if there really was some other error returned from MC